From db9a3bf0d7d14d945b9a371b485ecadba81a5919 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@collabora.com>
Date: Fri, 11 Apr 2014 12:31:29 -0400
Subject: [PATCH] fdo#75970: Let's do it in a straightforward fashion.

No idea what the intent of this weird static heap instantiation, but
apparently it's not helping us.

Change-Id: If6cccf649958b6cf01ede992c286af68eaeb31f6
(cherry picked from commit 0ffe4358c755504bc2754ef88edadde33b5f395a)
Reviewed-on: https://gerrit.libreoffice.org/8939
Tested-by: Markus Mohrhard <markus.mohrhard@googlemail.com>
Reviewed-by: Markus Mohrhard <markus.mohrhard@googlemail.com>
---
 sc/source/ui/view/viewfun2.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sc/source/ui/view/viewfun2.cxx b/sc/source/ui/view/viewfun2.cxx
index 800523e..241d5dd 100644
--- a/sc/source/ui/view/viewfun2.cxx
+++ b/sc/source/ui/view/viewfun2.cxx
@@ -1638,8 +1638,8 @@ void ScViewFunc::SearchAndReplace( const SvxSearchItem* pSearchItem,
 
             if (nCommand == SVX_SEARCHCMD_FIND_ALL || nCommand == SVX_SEARCHCMD_REPLACE_ALL)
             {
-                static SearchResults *aSearchResults = new SearchResults(pDoc);
-                aSearchResults->Show(aMatchedRanges);
+                SearchResults aSearchResults(pDoc);
+                aSearchResults.Show(aMatchedRanges);
 
                 rMark.ResetMark();
                 for (size_t i = 0, n = aMatchedRanges.size(); i < n; ++i)
-- 
1.9.0

